extensions/ql-vscode/src/run-queries.ts

```diff
@@ -348,7 +348,7 @@ async function compileNonDestructiveUpgrade(
   if (!query.dbItem?.contents?.dbSchemeUri) {
     throw new Error('Database is invalid, and cannot be upgraded.');
   }
-  const { scripts, matchesTarget } = await qs.cliServer.resolveUpgrades(query.dbItem.contents.dbSchemeUri.fsPath, searchPath, query.queryDbscheme);
+  const { scripts, matchesTarget } = await qs.cliServer.resolveUpgrades(query.dbItem.contents.dbSchemeUri.fsPath, searchPath, true, query.queryDbscheme);
```

Comment:
It's unclear to me why this is the only place where we allow a downgrade. Can you clarify?

Reply:
We only want to downgrade with non-destructive cases. The other uses are either in the manual upgrade which is destructive or for suggesting destructive upgrades when running with older cli versions.